### PR TITLE
feat(gossip): new parquet file gossip crate

### PR DESCRIPTION
**Pull request by domodwyer**
Adds a new (currently unused) `gossip_parquet_file` crate that layers a parquet-specific API on top of the `gossip` transport crate.

The API is simple - users interested in knowing when a parquet file has been created implement this:

https://github.com/influxdata/influxdb_iox/blob/6a0c54758b117104d512d6287e3508c2eb4b2b30/gossip_parquet_file/src/rx.rs#L16-L21

And it's called each time a peer receives a gossip notification, providing the `ParquetFile` that was wrote. This notification is guaranteed to happen _after_ the catalog has been updated. It's not guaranteed to ever be received though.

I could tell you a joke about UDP, but you might not get it.

---

* refactor(gossip): define NewParquetFiles topic (04051aea4)
      
      Separate the parquet file gossiping onto a new topic to allow specific
      interest subscriptions and minimise cluster traffic.

* feat(gossip): new parquet file gossip crate (6a0c54758)
      
      Adds a reusable "gossip_parquet_file" crate that provides a use-case
      specific wrapper over the underlying gossip transport.
      
      This crate deals with the encoding and decoding of parquet gossip
      messages, handling them off to the application, and decoupling latency
      of handlers from the gossip reactor.

### CODE REVIEW

File: gossip_parquet_file/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+//! ```
+//!
+//! # Best Effort
+//!
+//! This underlying gossip subsystem is designed to provide best effort delivery
+//! of messages, and therefore best-effort delivery of parquet creation events,
+//! without any ordering or delivery guarantees.
+//!
+//! This crate does NOT provide any eventual consistency guarantees.
+//!
```
Comment (domodwyer) [inline gossip_parquet_file/src/lib.rs:57]: An overview of the various types and how this fits with the gossip library and caller/application :+1:

File: gossip_parquet_file/src/rx.rs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+use generated_types::prost::Message;
+use observability_deps::tracing::{info, warn};
+use tokio::{sync::mpsc, task::JoinHandle};
+
+/// A [`ParquetFile`] notification handler received via gossip.
+#[async_trait]
+pub trait ParquetFileEventHandler: Send + Sync + Debug {
+    /// Process `message`.
+    async fn handle(&self, event: ParquetFile);
+}
```
Comment (domodwyer) [inline gossip_parquet_file/src/rx.rs:21]: Callers interested in knowing when files are persisted implement this method.

File: gossip_parquet_file/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,225 @@
+//! Parquet file creation notifications over [gossip].
+//!
+//! This sub-system is composed of the following primary components:
+//!
+//! * [`gossip`] crate: provides the gossip transport, the [`GossipHandle`], and
+//!   the [`Dispatcher`]. This crate operates on raw bytes.
+//!
+//! * The outgoing [`ParquetFileTx`]: a schema-specific wrapper over the
```
Comment (savage-engineer) [inline gossip_parquet_file/src/lib.rs:8]: Should this be:

```suggestion
//! * The outgoing [`ParquetFileTx`]: a parquet file focussed wrapper over the
```

**Review by savage-engineer — APPROVED** [review]
Nice 👍

**Pull request merged**